HTML/html_completions.py

```diff
@@ -257,6 +263,39 @@ def get_completions(self, view, prefix, locations, is_inside_tag):
 
         return (completion_list, flags)
 
+    def entity_completion_list(self):
+        completions = [
+            sublime.CompletionItem(
+                trigger='#00;',
+                completion='#${1:00};',
+                completion_format=sublime.COMPLETION_FORMAT_SNIPPET,
+                kind=(sublime.KIND_ID_MARKUP, 'e', 'Entity'),
+                details='Base-10 Unicode character',
+            ),
+            sublime.CompletionItem(
+                trigger='#x0000;',
+                completion='#x${1:0000};',
+                completion_format=sublime.COMPLETION_FORMAT_SNIPPET,
+                kind=(sublime.KIND_ID_MARKUP, 'e', 'Entity'),
+                details='Base-16 Unicode character',
+            ),
+        ]
+
+        completions += [
+            sublime.CompletionItem(
+                trigger=entity,
+                annotation=printed,
+                completion=entity,
+                completion_format=sublime.COMPLETION_FORMAT_TEXT,
+                kind=(sublime.KIND_ID_MARKUP, 'e', 'Entity'),
+                details=f'Renders as <code>{printed}</code>',
+            )
+            for entity, printed in html.entities.html5.items()
+            if entity.endswith(';')
+        ]
+
+        return completions
+
```

Comment:
Doesn't make a big difference, but how about

```python
    def entity_completion_list(self):
        return [
            sublime.CompletionItem(
                trigger='#00;',
                completion='#${1:00};',
                completion_format=sublime.COMPLETION_FORMAT_SNIPPET,
                kind=(sublime.KIND_ID_MARKUP, 'e', 'Entity'),
                details='Base-10 Unicode character',
            ),
            sublime.CompletionItem(
                trigger='#x0000;',
                completion='#x${1:0000};',
                completion_format=sublime.COMPLETION_FORMAT_SNIPPET,
                kind=(sublime.KIND_ID_MARKUP, 'e', 'Entity'),
                details='Base-16 Unicode character',
            ),
        ] + [
            sublime.CompletionItem(
                trigger=entity,
                annotation=printed,
                completion=entity,
                completion_format=sublime.COMPLETION_FORMAT_TEXT,
                kind=(sublime.KIND_ID_MARKUP, 'e', 'Entity'),
                details=f'Renders as <code>{printed}</code>',
            )
            for entity, printed in html.entities.html5.items()
            if entity.endswith(';')
        ]
```

Reply:
By defining a class-level or global attribute for `(sublime.KIND_ID_MARKUP, 'e', 'Entity')` the time needed to compute completions could be slightly reduced.

```python
class HtmlTagCompletions(sublime_plugin.EventListener):
    """
    Provide tag completions for HTML
    It matches just after typing the first letter of a tag name
    """

    KIND_ENTITY = (sublime.KIND_ID_MARKUP, 'e', 'Entity')
    ...

Reply:
Then I want to suggest `return `["#00;", "#x0000;", *(comprehension)]` for  nice optimizations.